src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -1777,6 +1828,19 @@ def update_vm_properties(self, agentpool: AgentPool) -> AgentPool:
             agentpool.mode = mode
         return agentpool
 
+    def update_network_profile(self, agentpool: AgentPool) -> AgentPool:
+        self._ensure_agentpool(agentpool)
+
+        asg_ids = self.context.get_asg_ids()
+        allowed_host_ports = self.context.get_allowed_host_ports()
+        if asg_ids or allowed_host_ports:
+            agentpool.network_profile = self.models.AgentPoolNetworkProfile()
```

Comment:
This could overwrite existing fields in network_profile by creating a new one. I will fix it in a following PR.